src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/PlatformManager.java

```diff
@@ -0,0 +1,217 @@
+        return Optional.ofNullable(platforms.get(name));
+    }
+
+    /**
+     * Will return the Worker interface which should be used to identify workers for the given platform
+     *
+     * @param name The name of the platform
+     * @return The interface used to identify a worker
+     */
+    public Optional<Worker> getWorker(String name) {
+        return getPlatform(name).map(platform -> platform.getWorker().orElse(fallbackWorker));
+    }
+
+    /**
+     * Will return the payment service which should be used for the given platform
+     * If there is no Platform with the given name None is returned.
+     *
+     * @param name The name of the platform to use
+     * @return The interface used for payment
+     */
+    public Optional<Payment> getPlatformPayment(String name) {
+        return getPlatform(name).map(platform -> platform.getPayment().orElse(fallbackPayment));
+    }
+
+    /**
+     * Publish the given experiment on the platform.
+     * The method will update the database with the new public task
+     *
+     * @param name The name of the platform
+     * @param experiment The experiment to publish
+     * @return None if the platform does not exist
+     */
+    public Optional<CompletableFuture<Boolean>> publishTask(String name, Experiment experiment) throws TaskOperationException {
+        if (tasksOps.searchTask(name,experiment.getId()).isPresent())
+            throw new TaskOperationException("Experiment is already published!");
+        return getPlatform(name).
+                map(platform1 -> platform1.publishTask(experiment)).
+                map(stringCompletableFuture -> stringCompletableFuture.handle((s, throwable) -> {
+                    if (s != null && throwable == null) {
+                        TaskRecord record = new TaskRecord();
+                        record.setExperiment(experiment.getId());
+                        record.setPlatformData(s);
+                        record.setStatus(TaskStatus.running);
+                        record.setCrowdPlatform(name);
+                        tasksOps.createTask(record);
+                    }
+                    return true;
+                }));
+    }
+
+    /**
+     * Unpublish a given experiment from the given platform
+     *
+     * @param name The name of the platform
+     * @param experiment The experiment to unpublish
+     * @return None if the platform was not found, false if the unpublish failed and true if everything went fine
+     */
+    public Optional<CompletableFuture<Boolean>> unpublishTask(String name, Experiment experiment) throws TaskOperationException {
+        TaskRecord record;
+
+        record = tasksOps.searchTask(name, experiment.getId()).
+                orElseThrow(() -> new TaskOperationException("Experiment is not published"));
+
+        return getPlatform(name).map(platform1 ->
+            platform1.unpublishTask(record.getPlatformData()).handle((b, throwable) -> {
```

Comment:
if you only need the result and not the error (if there was one), i would suggest thenApply